backend/src/models/reports/Tab_12_rpt_CF_InvoicePaymentsbyPortfolio.js

```diff
@@ -1,16 +1,239 @@
+            NULL total_expense_amount
+          FROM
+            contract_resource
+          GROUP BY
+            contract_id,
+            fiscal
+          UNION
+          SELECT
+            contract_id,
+            fiscal,
+            Sum(
+              CASE
+                WHEN is_expense = false THEN deliverable_amount
+                ELSE Cast(0 AS MONEY)
+              END ),
+            Sum(
+              CASE
+                WHEN is_expense = true THEN deliverable_amount
+                ELSE Cast(0 AS MONEY)
+              END )
+          FROM
+            contract_deliverable
+          GROUP BY
+            contract_id,
+            fiscal) t_sub
+          GROUP BY
+            contract_id,
+            fiscal ) t ON c.id = t.contract_id
+      LEFT JOIN  fiscal_year fy ON t.fiscal = fy.id
+      LEFT JOIN  project p ON c.project_id = p.id
+      LEFT JOIN  lateral (
+        SELECT
+          portfolio_id
+        FROM
+          sid_internal_coding
+        WHERE contract_id = c.id limit 1) sic ON true
+      LEFT JOIN  portfolio po ON po.id = sic.portfolio_id) AS q1`
+  ),
+
+  q2: knex.raw(
+    `(
+      SELECT
+        c.id contract_id,
+        c.co_number,
+        i.fiscal,
+        fy.fiscal_year,
+        i.invoice_date,
+        i.billing_period,
+        i.invoice_number,
+        sum( id.rate * id.unit_amount ) amount,
+        c.status,
+        s.supplier_name
+      FROM
+        contract c
+      LEFT JOIN invoice i ON c.id = i.contract_id
+      LEFT JOIN fiscal_year fy ON i.fiscal = fy.id
+      LEFT JOIN invoice_detail id ON i.id = id.invoice_id
+      LEFT JOIN supplier s ON c.supplier_id = s.id
+      GROUP BY
+        c.id,
+        c.co_number,
+        i.fiscal,
+        fy.fiscal_year,
+        i.invoice_date,
+        i.billing_period,
+        i.invoice_number,
+        c.status,
+        s.supplier_name) AS q2`
+  ),
+};
+
+/**
+ * Retrieves the final queries for invoice payments, filtered by fiscal year
+ *
+ * @param   {number | string | Array} fiscal- The fiscal year to grab data for
+ * @returns {Promise}                         - A promise that resolves to the query result
+ */
+const reportQueries = {
+  fiscalYear: (fiscal) =>
+    knex("fiscal_year").select("fiscal_year").where("fiscal_year.id", fiscal).first(),
+
+  // The columns on which to calculate totals.
+  columns: {
+    total_fee_amount: "total_fee_amount",
+    total_expense_amount: "total_expense_amount",
+    total_remaining: "total_remaining",
+    apr: "apr",
+    may: "may",
+    jun: "jun",
+    jul: "jul",
+    aug: "aug",
+    sep: "sep",
+    oct: "oct",
+    nov: "nov",
+    dec: "dec",
+    jan: "jan",
+    feb: "feb",
+    mar: "mar",
+    remaining: "remaining",
+    descoped: "descoped",
+  },
+
+  // The report query, which builds off of the base queries.
+  report: (fiscal) =>
+    knex(baseQueries.q1)
+      .select({
+        portfolio_name: "q1.portfolio_name",
```

Comment:
I might use this technique on my report, once the query has been fixed by the DBA